tests/lib/test_docker.py

```diff
@@ -58,3 +59,35 @@ def run(args, timeout, **kwargs):
     monkeypatch.setattr(docker, "subprocess_run", error(None, msg))
     with pytest.raises(docker.DockerDiskSpaceError):
         docker.docker([])
+
+
+@pytest.mark.needs_docker
+def test_copy_from_volume(tmp_path, docker_cleanup):
+    volume = __name__
+    src = tmp_path / "src.txt"
+    dst = tmp_path / "dst.txt"
+    tmp = dst.with_suffix(dst.suffix + ".tmp")
+    src.write_text("I exist")
+    docker.create_volume(volume)
+    docker.copy_to_volume(volume, src, "src.txt")
+    docker.copy_from_volume(volume, "src.txt", dst)
+    assert dst.read_text() == "I exist"
+    assert not tmp.exists()
+
+
+@pytest.mark.needs_docker
+@pytest.mark.skipif(not Path("/dev/full").exists(), reason="No /dev/full, skipping")
+def test_copy_from_volume_error(tmp_path, docker_cleanup):
+    volume = __name__
+    src = tmp_path / "src.txt"
+    dst = tmp_path / "dst.txt"
+    tmp = dst.with_suffix(dst.suffix + ".tmp")
+
+    src.write_text("I exist")
+    docker.create_volume(volume)
+    docker.copy_to_volume(volume, src, "src.txt")
+    with pytest.raises(Exception):
+        docker.copy_from_volume(volume, "src.txt", "/dev/full")
+        
+    assert not dst.exists()
```

Comment:
I have realised that this assertions is kinda pointless, as it will never exist as we pass in /dev/null